### PR TITLE
Compressed images on the teams page

### PR DESCRIPTION
**Pull request by Ruxuan**
Compressed W19 and F18 headshots on the teams page using python script provided in PR.

Sample bandwidth savings,
W19 Headshots folder size: **59.4 MB**
W19 Headshots Resized Compressed folder size: **2.8 MB**

Remaining uncompressed images are members in the alumni section which will be paginated and only loaded on user action rather than page load.

All images are resized to 500x500 because there's no noticeable difference when presented on the website. See a sample comparison below,
screenshot 2019-01-20 00 23 59
screenshot 2019-01-20 00 24 07

### CODE REVIEW

**Comment by imruahmed** [comment]
resizing to 500x500 will mess with the aspect ratio for some pictures -- can we figure out another way to get pictures to the right size?

File: tools/resize_and_compress.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import sys
+import os
+from PIL import Image, ImageOps
+
+if len(sys.argv) < 2:
+  print("Missing image directory argument")
+  sys.exit(1)
+
+OUTPUT_DIR = "./images"
```
Comment (imruahmed) [inline tools/resize_and_compress.py]: is this local? and then u uploaded all the pictures by hand?

**Comment by Ruxuan** [comment]
The Pillow library has this weird ImageOps.fit() command that keeps original aspect ratio. Sometimes the sides are cut off a little bit.

File: tools/resize_and_compress.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  os.mkdir(OUTPUT_DIR)
+
+IMAGE_SIZE = 500
+image_directory = sys.argv[1]
+
+for filename in os.listdir(image_directory):
+  if (filename.endswith(".jpg") or filename.endswith(".png") or
+      filename.endswith(".jpeg") or filename.endswith(".JPG")):
+    print("Resizing and compressing: " + filename)
+    img = Image.open(image_directory + "/" + filename)
```
Comment (imruahmed) [inline tools/resize_and_compress.py]: use `os.path.join` instead -- https://docs.python.org/3/library/os.path.html#os.path.join

File: tools/resize_and_compress.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    width, height = img.size
+
+    image_size = (min(width, IMAGE_SIZE), min(height, IMAGE_SIZE))
```
Comment (imruahmed) [inline tools/resize_and_compress.py]: i left a comment already about doing this differently. i think we can maintain the aspect ratio by scaling the width or height to 500px and scaling the other accordingly

File: tools/resize_and_compress.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  sys.exit(1)
+
+OUTPUT_DIR = "./images"
+if not os.path.exists(OUTPUT_DIR):
+  os.mkdir(OUTPUT_DIR)
+
+IMAGE_SIZE = 500
+image_directory = sys.argv[1]
+
+for filename in os.listdir(image_directory):
```
Comment (imruahmed) [inline tools/resize_and_compress.py]: does this mean you downloaded all the pictures? would it be better if we got the pictures directly from firebase and then reuploaded them

File: tools/resize_and_compress.py (hunk 1 above)
Comment (Ruxuan) [reply tools/resize_and_compress.py]: Yea it outputs to local directory. You can upload a folder to firebase pretty easily through their web ui.

File: tools/resize_and_compress.py (hunk 4 above)
Comment (Ruxuan) [reply tools/resize_and_compress.py]: I had all the picture on my computer from the first time I transferred them to firebase. For the future, I don't think we'll want to even upload non-compressed images to firebase so there won't be a need to get pictures from firebase and reupload them. I'm looking to delete the full sized pictures off firebase after we switch master branch to the compressed ones.

File: tools/resize_and_compress.py (hunk 2 above)
Comment (Ruxuan) [reply tools/resize_and_compress.py]: 👍

**Pull request merged**